src/Vision/Annotation.php

```diff
@@ -0,0 +1,298 @@
+<?php
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Vision;
+
+use Google\Cloud\Vision\Annotation\Entity;
+use Google\Cloud\Vision\Annotation\Face;
+use Google\Cloud\Vision\Annotation\ImageProperties;
+use Google\Cloud\Vision\Annotation\SafeSearch;
+
+/**
+ * Represents a [Google Cloud Vision](https://cloud.google.com/vision) image
+ * annotation result.
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\ServiceBuilder;
+ *
+ * $cloud = new ServiceBuilder();
+ * $vision = $cloud->vision();
+ *
+ * $imageResource = fopen(__DIR__ .'/assets/family-photo.jpg', 'r');
+ * $image = $vision->image($imageResource, [
+ *     'FACE_DETECTION'
+ * ]);
+ *
+ * $annotation = $vision->annotate($image);
+ * ```
+ */
+class Annotation
+{
+    /**
+     * @var array
+     */
+    private $info;
+
+    /**
+     * @var array
+     */
+    private $faces;
+
+    /**
+     * @var array
+     */
+    private $landmarks;
+
+    /**
+     * @var array
+     */
+    private $logos;
+
+    /**
+     * @var array
+     */
+    private $labels;
+
+    /**
+     * @var array
+     */
+    private $text;
+
+    /**
+     * @var SafeSearch
+     */
+    private $safeSearch;
+
+    /**
+     * @var ImageProperties
+     */
+    private $imageProperties;
+
+    /**
+     * @var array
+     */
+    private $error;
+
+    /**
+     * Create an annotation result.
+     *
+     * This class represents a single image annotation response from Cloud Vision. If multiple images were tested at
+     * once, the result will be an array of Annotation instances.
+     *
+     * This class is not intended to be instantiated outside the gcloud-php library.
+     *
+     * @param array $info The annotation result
+     */
+    public function __construct($info)
+    {
+        $this->info = $info;
+
+        if (isset($info['faceAnnotations'])) {
+            $this->faces = [];
```

Comment:
I think I would prefer to always return an array, but that may just be me :). If we do continue to return null in these scenarios we should at least document that in the return block for `$annotation->landmarks()` and the similar methods.